Foundation/NSError.swift

```diff
@@ -244,7 +445,909 @@ public extension __BridgedNSError where RawValue: UnsignedInteger {
+///
+/// This protocol is used primarily to generate the conformance to
+/// _ObjectTypeBridgeableError for such an enum.
+public protocol _BridgedNSError : __BridgedNSError, RawRepresentable, _ObjectTypeBridgeableError, Hashable {
+    /// The NSError domain to which this type is bridged.
+    static var _nsErrorDomain: String { get }
+}
+
+/// Describes a bridged error that stores the underlying NSError, so
+/// it can be queried.
+public protocol _BridgedStoredNSError : __BridgedNSError, _ObjectTypeBridgeableError, CustomNSError, Hashable {
+    /// The type of an error code.
+    associatedtype Code: _ErrorCodeProtocol
+
+    /// The error code for the given error.
+    var code: Code { get }
+
+    //// Retrieves the embedded NSError.
+    var _nsError: NSError { get }
+
+    /// Create a new instance of the error type with the given embedded
+    /// NSError.
+    ///
+    /// The \c error must have the appropriate domain for this error
+    /// type.
+    init(_nsError error: NSError)
+}
+
+/// Various helper implementations for _BridgedStoredNSError
+public extension _BridgedStoredNSError where Code: RawRepresentable, Code.RawValue: SignedInteger {
+    // FIXME: Generalize to Integer.
+    public var code: Code {
+        return Code(rawValue: numericCast(_nsError.code))!
+    }
+
+    /// Initialize an error within this domain with the given ``code``
+    /// and ``userInfo``.
+    public init(_ code: Code, userInfo: [String : Any] = [:]) {
+        self.init(_nsError: NSError(domain: Self._nsErrorDomain,
+            code: numericCast(code.rawValue),
+            userInfo: userInfo))
+    }
+
+    /// The user-info dictionary for an error that was bridged from
+    /// NSError.
+    var userInfo: [String : Any] { return errorUserInfo }
+}
+
+/// Various helper implementations for _BridgedStoredNSError
+public extension _BridgedStoredNSError where Code: RawRepresentable, Code.RawValue: UnsignedInteger {
+    // FIXME: Generalize to Integer.
+    public var code: Code {
+        return Code(rawValue: numericCast(_nsError.code))!
+    }
+
+    /// Initialize an error within this domain with the given ``code``
+    /// and ``userInfo``.
+    public init(_ code: Code, userInfo: [String : Any] = [:]) {
+        self.init(_nsError: NSError(domain: Self._nsErrorDomain,
+            code: numericCast(code.rawValue),
+            userInfo: userInfo))
+    }
+}
+
+/// Implementation of __BridgedNSError for all _BridgedStoredNSErrors.
+public extension _BridgedStoredNSError {
+    /// Default implementation of ``init(_bridgedNSError)`` to provide
+    /// bridging from NSError.
+    public init?(_bridgedNSError error: NSError) {
+        if error.domain != Self._nsErrorDomain {
+            return nil
+        }
+
+        self.init(_nsError: error)
+    }
+}
+
+/// Implementation of CustomNSError for all _BridgedStoredNSErrors.
+public extension _BridgedStoredNSError {
+    // FIXME: Would prefer to have a clear "extract an NSError
```

Comment:
Oops, we should kill this FIXME in the overlay; this is what `getEmbeddedNSError` does.